core/commands/pull/interactors/fetch_pull_request.py

```diff
@@ -1,9 +1,24 @@
+from datetime import UTC, datetime, timedelta
+
+from shared.django_apps.core.models import Pull
+
 from codecov.commands.base import BaseInteractor
 from codecov.db import sync_to_async
-from core.models import Pull
+from services.task.task import TaskService
 
 
 class FetchPullRequestInteractor(BaseInteractor):
+    def _should_sync_pull(self, pull: Pull | None) -> bool:
+        return (
+            pull is not None
+            and pull.state == "open"
+            and (datetime.now(tz=None) - pull.updatestamp) > timedelta(hours=1)
+        )
+
     @sync_to_async
     def execute(self, repository, id):
-        return repository.pull_requests.filter(pullid=id).first()
+        pull = repository.pull_requests.filter(pullid=id).first()
+        if self._should_sync_pull(pull):
+            TaskService().pulls_sync(repository.repoid, id)
```

Comment:
Could the pulls_sync task leave the pull syncing indefinitely? I suppose there's a hard timeout, but trying to think if this could end up in another "stale" state

Reply:
Well even if the task does have an issue it doesn't matter for this request in particular because the calling of it only enqueues it for the workers. It's not blocking in any way.